OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/FieldAccessInformationAnalysis.scala

```diff
@@ -164,18 +202,47 @@ object EagerFieldAccessInformationAnalysis extends BasicFPCFEagerAnalysisSchedul
         MethodFieldWriteAccessInformation
     )
 
-    override def derivesEagerly: Set[PropertyBounds] = Set.empty
-
-    override def derivesCollaboratively: Set[PropertyBounds] = PropertyBounds.ubs(
+    protected def derivedProperties: Set[PropertyBounds] = PropertyBounds.ubs(
         FieldReadAccessInformation,
         FieldWriteAccessInformation,
         MethodFieldReadAccessInformation,
         MethodFieldWriteAccessInformation
     )
+}
+
+object EagerFieldAccessInformationAnalysis
+    extends FieldAccessInformationAnalysisScheduler
+    with BasicFPCFEagerAnalysisScheduler {
+
+    override def requiredProjectInformation: ProjectInformationKeys = super.requiredProjectInformation ++ Seq(
+        DeclaredMethodsKey,
+        ContextProviderKey
+    )
+
+    override def derivesEagerly: Set[PropertyBounds] = Set.empty
+
+    override def derivesCollaboratively: Set[PropertyBounds] = derivedProperties
```

Comment:
Why do you need to override these, can't you specify them in the super trait already? I wouldn't expect us to ever add a lazy scheduler where they wouldn't fit.

Reply:
Since these functions stem from the specific `BasicEagerFPCFAnalysisScheduler` and `BasicTriggeredFPCFAnalysisScheduler` traits and not from the generic `FPCFAnalysisScheduler` trait that the super trait of these extends, they are correct here. It also only makes sense that you can specify e.g. eager behaviour when you know you have an eager analysis.

Reply:
No, they stem from `ComputationSpecification`, the super trait of `FPCFAnalysisScheduler`. It's fine like this, though.

Reply:
It seems the Github UI is not fit for determining the origin of super functions...